kolibri/plugins/coach/assets/src/views/plan/GroupsPage/CreateGroupModal.vue

```diff
@@ -82,21 +82,18 @@
       },
     },
     methods: {
-      ...mapActions('groups', ['displayModal', 'createGroup']),
+      ...mapActions('groups', ['createGroup']),
       callCreateGroup() {
         this.formSubmitted = true;
         if (this.formIsValid) {
           this.submitting = true;
           this.createGroup({ groupName: this.name, classId: this.classId }).then(() => {
-            this.$emit('success');
+            this.$emit('submit');
```

Comment:
won't this emit submit twice? Once for `v-on="$listeners"` and once here?

Reply:
CreateGroupModal only emits 'submit' once. I think if you explicitly set a `@submit=customListener`, it sort of overwrites the `{submit: parentListener}` hidden in `v-on="$listeners`

Reply:
Interestingly, though, if you remove this `$emit`, the event from CreateGroupModal does not appear in the devtools, but it seems that it still goes through the GroupsPage.

Reply:
yeah that seems reasonable.

another alternative is for modals that handle their own internal async code, to explicitly emit an event that's _not_ `submit` or `cancel`. For example:

```javascript
.then(() => this.$emit('finally'))
```